cpp/modules/deck.gl/layers/src/solid-polygon-layer/solid-polygon-layer.cpp

```diff
@@ -224,3 +243,86 @@ auto SolidPolygonLayer::getLineColorData(const std::shared_ptr<arrow::Table>& ta
 
   return ArrowMapper::mapVector4FloatColumn(table, props->getLineColor);
 }
+
+auto SolidPolygonLayer::_getModels(wgpu::Device device) -> std::list<std::shared_ptr<lumagl::Model>> {
+  // JS returns both models for the top and side if properties are set as such, so return a list
+  // of models with id = either 'id-top' or 'id-side'
+  std::string id = this->props()->id;
+  bool filled = this->props()->filled;
+  bool extruded = this->props()->extruded;
+
+  std::list<std::shared_ptr<lumagl::Model>> models_list;
+
+  if (filled) {
+    // make new model using vsTop
+    // TODO(randy@unfolded.ai): Remove instanced fields, add (wireframe, isSideVertex) uniforms to top
+    std::vector<std::shared_ptr<garrow::Field>> attributeFields{
+        std::make_shared<garrow::Field>("vertexPositions", wgpu::VertexFormat::Float2),
+        std::make_shared<garrow::Field>("vertexValid", wgpu::VertexFormat::Float),
+        std::make_shared<garrow::Field>("positions", wgpu::VertexFormat::Float3),
+        std::make_shared<garrow::Field>("elevations", wgpu::VertexFormat::Float),
+        std::make_shared<garrow::Field>("fillColors", wgpu::VertexFormat::Float4),
+        std::make_shared<garrow::Field>("lineColors", wgpu::VertexFormat::Float4),
+        std::make_shared<garrow::Field>("pickingColors", wgpu::VertexFormat::Float3)};
+
+    auto attributeSchema = std::make_shared<lumagl::garrow::Schema>(attributeFields);
+
+    // TODO(randy@unfolded.ai): Although instanced fields don't seem to be used in the JS vertex-top shader,
+    //                          leaving as nullptr causes a segmentation fault. Using placeholder for now.
+    std::vector<std::shared_ptr<garrow::Field>> instancedFields{
+        std::make_shared<garrow::Field>("placeholder", wgpu::VertexFormat::Float)};
```

Comment:
See comment above. Seems that setting this to `nullptr` causes a segmentation fault. In the JS code, `isInstanced`, passed to the Model constructor, is not set to 1/true. So it seems we don't want instanced parameters like what is in `if (extruded) {` below?

Reply:
I believe we are using two table here, one contains instanced attributes and one contains non-instanced attributes.

Some layers do not use instanced attributes. I think the polygon layer does not use instanced attributes.

You need to make sure the call to pipeline->Draw inside Model has 0 instance count parameter.